internal/shared/util/image/pull.go

```diff
@@ -224,6 +224,11 @@ func (p *ContainersImagePuller) applyImage(ctx context.Context, ownerID string,
 		}
 	}()
 
+	if hasChart(img) {
+		return pullChart(ctx, ownerID, imgSrc, canonicalRef, cache, layoutDir)
+	}
+
+	// Helm charts would error when getting OCI config
```

Comment:
This tells me that our cache `Store` interface method is too specific. We need to make that generic enough to accommodate registry+v1 bundles and OCI helm charts as a first step, and rebase the feature-gated helm support on top.

Reply:
Making the `Cache` interface generic enough to be able to store both Helm charts and OCI images is inevitable. Will be looking at that at some point

Reply:
@camilamacedo86 all concerns raised should be resolved.